test/unit/tasks/publishing_api_test.rb

```diff
@@ -0,0 +1,384 @@
+require "test_helper"
+
+class PublishingApiRake < ActiveSupport::TestCase
+  extend Minitest::Spec::DSL
+
+  teardown do
+    task.reenable if defined?(task) # without this, calling `invoke` does nothing after first test
+  end
+
+  describe "#publish_special_routes" do
+    let(:task) { Rake::Task["publishing_api:publish_special_routes"] }
+
+    test "publishes each special route" do
+      Timecop.freeze do
+        params = {
+          format: "special_route",
+          publishing_app: "whitehall",
+          rendering_app: Whitehall::RenderingApp::WHITEHALL_FRONTEND,
+          update_type: "major",
+          type: "prefix",
+          public_updated_at: Time.zone.now.iso8601,
+        }
+
+        SpecialRoute.all.each do |route|
+          GdsApi::PublishingApi::SpecialRoutePublisher
+            .any_instance.expects(:publish).with(params.merge(route))
+        end
+
+        task.invoke
+      end
+    end
+  end
+
+  describe "#publish_redirect_routes" do
+    let(:task) { Rake::Task["publishing_api:publish_redirect_routes"] }
+
+    test "publishes each redirect route" do
+      Timecop.freeze do
+        RedirectRoute.all.each do |route|
+          params = {
+            base_path: route[:base_path],
+            document_type: "redirect",
+            schema_name: "redirect",
+            locale: "en",
+            details: {},
+            redirects: [
+              {
+                path: route[:base_path],
+                type: route.fetch(:type, "prefix"),
+                destination: route[:destination],
+              },
+            ],
+            publishing_app: "whitehall",
+            public_updated_at: Time.zone.now.iso8601,
+            update_type: "major",
+          }
+
+          GdsApi::PublishingApi
+            .any_instance.expects(:put_content).with(route[:content_id], params)
+
+          GdsApi::PublishingApi
+            .any_instance.expects(:publish).with(route[:content_id])
+        end
+
+        task.invoke
+      end
+    end
+  end
+
+  describe "republish" do
+    describe "#*_by_slug" do
+      tasks = [
+        { name: "organisation_by_slug", model: "Organisation" },
+        { name: "person_by_slug", model: "Person" },
+        { name: "role_by_slug", model: "Role" },
+      ]
+
+      test "republishes record by slug" do
+        tasks.each do |task|
+          record = create(task[:model].underscore) # rubocop:disable Rails/SaveBang
+          model = task[:model].constantize
```

Comment:
The more conventional approach to this kind of DRYing up of tests is to use [shared example groups](https://relishapp.com/rspec/rspec-core/v/2-0/docs/example-groups/shared-example-group).

Reply:
I wish this was RSpec world, unfortunately we're currently faking RSpec features in Minitest which doesn't include shared examples, I don't think.

Reply:
Ah yes, I'd missed that this is Minitest.